README.md

```diff
@@ -137,6 +137,99 @@ client.link_to(stream: 'anotherstream', events: [exisiting_event1, ...])
 When you read from stream where links are placed. By default Event Store Client always resolve links for you returning the event that points to the link. You can use the ES-ResolveLinkTos: false HTTP header during readin stream to tell Event Store Client to return you the actual link and to not resolve it.
 More info: [ES-ResolveLinkTos](https://eventstore.org/docs/http-api/optional-http-headers/resolve-linkto/index.html?tabs=tabid-1%2Ctabid-3).
 
+## Event Mappers
+
+We offer two types of mappers - default and encrypted.
+
+### Default Mapper
+
+This is used out of the box. It just translates the EventClass defined in your application to
+Event parsable by event_store and the other way around.
+
+### Encrypted Mapper
+
+This is implemented to match GDPR requirements. It allows you to encrypt any event using your
+encryption_key repository.
+
+```ruby
+mapper = EventStoreClient::Mapper::Encrypted.new(key_repository)
+EventStoreClient.configure do |config|
+  config.mapper = mapper
+end
+```
+
+The Encrypted mapper uses the encryption key repository to encrypt data in your events according to the event definition.
+
+Here is the minimal repository interface for this to work.
+
+```ruby
+class DummyRepository
+  class Key
+    attr_accessor :iv, :cipher, :id
+    def initialize(id:, **)
+      @id = id
+    end
+  end
+
+  def find(user_id)
+    Key.new(id: user_id)
+  end
+
+  def encrypt(*)
+    'darthvader'
+  end
+
+  def decrypt(*)
+    { first_name: 'Anakin', last_name: 'Skylwalker'}
+  end
+end
+```
+
+Now, having that, you only need to define the event encryption schema:
+
+```ruby
+class EncryptedEvent < EventStoreClient::DeserializedEvent
+  def schema
+    Dry::Schema.Params do
+      required(:user_id).value(:string)
+      required(:first_name).value(:string)
+      required(:last_name).value(:string)
+      required(:profession).value(:string)
+    end
+  end
+
+  def self.encryption_schema
+    {
+      key: ->(data) { data['user_id'] },
+      attributes: %i[first_name last_name email]
+    }
+  end
+end
+
+event = EncryptedEvent.new(
+  user_id: SecureRandom.uuid,
+  first_name: 'Anakin',
+  last_name: 'Skylwalker',
+  profession: 'Jedi'
+)
+```
+
+When you'll publish this event, in the store will be saved:
+
+```ruby
+{
+  'data' => {
+    'user_id' => 'dab48d26-e4f8-41fc-a9a8-59657e590716',
+    'first_name' => 'encrypted',
+    'last_name' => 'encrypted',
+    'profession' => 'Jedi',
+    'encrypted' => '2345l423lj1#$!lkj24f1'
```

Comment:
So this will contain both first name and last name? And why did you do it like this and not simply encrypt fields individually and write the encrypted value straight in the field?

Reply:
Performance. Your suggestion was my first implementation, but after discussing with @aroeczek we figured out this approach. 

Now you have 1 request to the encryption service per event to encrypt - in the old way, we had 1 request per each attribute to encrypt.